src/extension-consumer/src/produce/mod.rs

```diff
@@ -1,149 +1,269 @@
 use std::fs::File;
 use std::io::{BufReader, BufRead};
-use std::path::PathBuf;
+use std::path::{PathBuf, Path};
 use structopt::StructOpt;
+use tracing::debug;
 
 use fluvio::{Fluvio, TopicProducer};
 use fluvio_types::print_cli_ok;
 use crate::common::FluvioExtensionMetadata;
-use crate::Result;
-
-/// Produce log configuration parameters
-#[derive(Debug)]
-pub struct ProduceLogConfig {
-    pub topic: String,
-    pub partition: i32,
-    pub continuous: bool,
-}
-
-#[derive(Debug)]
-pub enum FileRecord {
-    Lines(PathBuf),
-    Files(Vec<PathBuf>),
-}
+use crate::{Result, ConsumerError};
 
 // -----------------------------------
 // CLI Options
 // -----------------------------------
 
+/// Write messages to a topic/partition
+///
+/// When no '--files' are provided, the producer will read from 'stdin'
+/// and send each line of input as one record.
+///
+/// If one or more files are given with '--files', each file is sent as
+/// one entire record.
+///
+/// If '--key-separator' or '--json-path' are used, records are sent as
+/// key/value pairs, and the keys are used to determine which partition
+/// the records are sent to.
 #[derive(Debug, StructOpt)]
 pub struct ProduceLogOpt {
     /// The name of the Topic to produce to
     #[structopt(value_name = "topic")]
     pub topic: String,
 
-    /// The ID of the Partition to produce to
-    #[structopt(
-        short = "p",
-        long = "partition",
-        value_name = "integer",
-        default_value = "0"
-    )]
-    pub partition: i32,
-
-    /// Send messages in an infinite loop
-    #[structopt(short = "C", long = "continuous")]
-    pub continuous: bool,
-
-    /// Send each line of the file as its own Record
-    #[structopt(
-        short = "l",
-        long = "record-per-line",
-        value_name = "filename",
-        parse(from_os_str)
-    )]
-    record_per_line: Option<PathBuf>,
-
-    /// Send an entire file as a single Record
-    #[structopt(
-        short = "r",
-        long = "record-file",
-        value_name = "filename",
-        parse(from_os_str),
-        conflicts_with = "record-per-line"
-    )]
-    record_file: Vec<PathBuf>,
+    /// Send each line of input as its own record (using '\n')
+    #[structopt(hidden = true, short, long)]
+    pub lines: bool,
```

Comment:
if we are going to support lines, let's rename this as `split` and support splitting arbitrary

Reply:
I actually just saw something that makes me unsure about doing this. I noticed that `BufReader::lines` actually will split on both `\n` and `\r\n`, meaning that there is not a single "split sequence", but two. I don't think there is an easy way to replicate that functionality via a "split" field in structopt.

I think this is something we may want to postpone until we have a better idea about what we want. I think, however, that we should leave this code as-is, because:

- If you look, this `--lines` field is marked as "hidden", meaning it does not show up in the help menu
- We do not document `--lines` so people will not be depending on it
- We can consider it an experiment as to whether it is useful:
  - If people ask for line-by-line functionality, we can tell them to try out the hidden `--lines` feature
  - If people ask for split-by functionality, we can ask them for their use-cases and come up with good requirements.

Reply:
That's fair comment.  Here's I think we should do.   Let's make lines as default since we need standard way to separate one record from another.   This way, we don't need  `--lines` option. We can bring back separator in the future.

Reply:
Right now, "line-by-line" is the default behavior for stdin, and "whole file is one record" is the default behavior for files. How does that sound?

Reply:
Binary files is no different from string just with different separator.  I think if we need to deal with binary file, we can provide file type option similar to consumer